tests/pytests/unit/modules/test_pcs.py

```diff
@@ -0,0 +1,314 @@
+            "pkg.version_cmp": MagicMock(return_value=1),
+        },
+    )
+
+    with patch_salt:
+        pcs.is_auth(test_data.nodes, pcsuser=test_data.username, pcspasswd=test_data.password)
+    
+    # print("Exp command: ", exp_cmd)
+    # print("call_args_list: ", mock_cmd.call_args_list[0][0][0] )
+    assert mock_cmd.call_args_list[0][0][0] == exp_cmd
+
+@pytest.mark.parametrize("ver_cmp_ret,old_ver", [(1, False), (0, True)])
+def test_cluster_setup(ver_cmp_ret, old_ver, test_data):
+    """
+    Test for seting up a cluster
+    """
+    exp_cmd = ["pcs", "cluster", "setup"]
+    if old_ver:
+        exp_cmd.append("--name")
+    exp_cmd = (
+        exp_cmd + [test_data.cluster_name] + test_data.nodes + test_data.extra_args
+    )
+
+    mock_cmd = MagicMock()
+    patch_salt = patch.dict(
+        pcs.__salt__,
+        {
+            "cmd.run_all": mock_cmd,
+            "pkg.version_cmp": MagicMock(return_value=ver_cmp_ret),
+        },
+    )
+    
+    with patch_salt:
+        pcs.cluster_setup(
+            test_data.nodes, test_data.cluster_name, extra_args=test_data.extra_args
+        )
+    assert mock_cmd.call_args_list[0][0][0] == exp_cmd
+
+def test_cluster_destroy(test_data):
+    """
+    Test for destroying a cluster
+    """
+    exp_cmd = ["pcs", "cluster", "destroy"]
+    exp_cmd.extend(test_data.extra_args)
+
+    mock_cmd = MagicMock()
+    patch_salt = patch.dict(pcs.__salt__, {"cmd.run_all": mock_cmd})
+    with patch_salt:
+        pcs.cluster_destroy(
+            extra_args=test_data.extra_args
+        )
+    
+    assert mock_cmd.call_args_list[0][0][0] == exp_cmd
+
+def test_cluster_node_add(test_data):
+    """
+    Test for adding a cluster
+    """
+    exp_cmd = ["pcs", "cluster", "node", "add"]
+    exp_cmd = exp_cmd + [test_data.nodea] + test_data.extra_args
+
+    mock_cmd = MagicMock()
+    patch_salt = patch.dict(pcs.__salt__, {"cmd.run_all": mock_cmd})
+    with patch_salt:
+        pcs.cluster_node_add(test_data.nodea, extra_args=test_data.extra_args)
+    assert mock_cmd.call_args_list[0][0][0] == exp_cmd
+
+def test_cib_push(test_data):
+    """
+    Test for pushing a CIB file
+    """
+    exp_cmd = ["pcs", "cluster", "cib-push", test_data.cib_filename, 'scope=configuration']
+    exp_cmd.extend(test_data.extra_args)
+
+    mock_cmd = MagicMock()
+    patch_salt = patch.dict(pcs.__salt__, {"cmd.run_all": mock_cmd})
+    with patch_salt:
+        pcs.cib_push(
+            test_data.cib_filename, scope="configuration", extra_args=test_data.extra_args
+            )
+    assert mock_cmd.call_args_list[0][0][0] == exp_cmd
+
+def test_config_show(test_data):
+    """
+    Test for config show
+    """
+    exp_cmd = ["pcs", "cluster", "cib-push", test_data.cib_filename, 'scope=configuration']
+    exp_cmd.extend(test_data.extra_args)
+
+    mock_cmd = MagicMock()
+    patch_salt = patch.dict(pcs.__salt__, {"cmd.run_all": mock_cmd})
+    with patch_salt:
+        pcs.cib_push(
```

Comment:
I'm not seeing where you are testing config show here.

Reply:
I removed it because a test on that is already covered with the show_items tests, This would just be testing a mock of that function.